### ISSUE TITLE
EDNS not working when deployed before TS

### ISSUE THREAD

**Issue by skenderidis** [F5Networks/k8s-bigip-ctlr#2796]
#### Setup Details
 CIS Version : 2.12   
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP 15.1   
 AS3 Version: 3.x      
 Agent Mode: AS3 (GTM with CCCL)       
 Orchestration: K8S     
 Orchestration Version: 1.20       
 Pool Mode: Cluster      
 Additional Setup details: Flannel

#### Description  
When deploying EDNS before the TS resources, then the GTM pools are with 0 pool members and are marked as red. 
If I deploy the same declarations but with different order (TS first), then everything works as normal.

#### Steps To Reproduce
1) Deploy the EDNS resources
```
apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-1
  labels:
    f5cr: "true"
spec:
  domainName: host1.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---
apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-2
  labels:
    f5cr: "true"
spec:
  domainName: host2.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---

apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-3
  labels:
    f5cr: "true"
spec:
  domainName: host3.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---

apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-4
  labels:
    f5cr: "true"
spec:
  domainName: host4.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---
apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-5
  labels:
    f5cr: "true"
spec:
  domainName: host5.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---


apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-6
  labels:
    f5cr: "true"
spec:
  domainName: host6.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---

apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-7
  labels:
    f5cr: "true"
spec:
  domainName: host7.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---

apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-8
  labels:
    f5cr: "true"
spec:
  domainName: host8.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---


apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-9
  labels:
    f5cr: "true"
spec:
  domainName: host9.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---
apiVersion: "cis.f5.com/v1"
kind: ExternalDNS
metadata:
  name: edns-app-10
  labels:
    f5cr: "true"
spec:
  domainName: host10.f5demo.cloud
  dnsRecordType: A
  loadBalanceMethod: round-robin
  pools:
  - dnsRecordType: A
    loadBalanceMethod: round-robin
    dataServerName: /Common/gtm-primary
---
```
2) Deploy the TS resources
```

apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-1
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.1"
  virtualServerPort: 80
  host: "host1.f5demo.cloud"
  virtualServerName: "app-1"
  mode: standard
  snat: auto
  pool:
    service: app-svc-1
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---

apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-2
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.2"
  virtualServerPort: 80
  host: "host2.f5demo.cloud"
  virtualServerName: "app-2"
  mode: standard
  snat: auto
  pool:
    service: app-svc-2
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---

apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-3
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.3"
  virtualServerPort: 80
  host: "host3.f5demo.cloud"
  virtualServerName: "app-3"
  mode: standard
  snat: auto
  pool:
    service: app-svc-3
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---


apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-4
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.4"
  virtualServerPort: 80
  host: "host4.f5demo.cloud"
  virtualServerName: "app-4"
  mode: standard
  snat: auto
  pool:
    service: app-svc-4
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---


apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-5
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.5"
  virtualServerPort: 80
  host: "host5.f5demo.cloud"
  virtualServerName: "app-5"
  mode: standard
  snat: auto
  pool:
    service: app-svc-5
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---

apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-6
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.6"
  virtualServerPort: 80
  host: "host6.f5demo.cloud"
  virtualServerName: "app-6"
  mode: standard
  snat: auto
  pool:
    service: app-svc-6
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---




apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-7
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.7"
  virtualServerPort: 80
  host: "host7.f5demo.cloud"
  virtualServerName: "app-7"
  mode: standard
  snat: auto
  pool:
    service: app-svc-7
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---
apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-8
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.8"
  virtualServerPort: 80
  host: "host8.f5demo.cloud"
  virtualServerName: "app-8"
  mode: standard
  snat: auto
  pool:
    service: app-svc-8
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---


apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-9
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.9"
  virtualServerPort: 80
  host: "host9.f5demo.cloud"
  virtualServerName: "app-9"
  mode: standard
  snat: auto
  pool:
    service: app-svc-9
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---
apiVersion: "cis.f5.com/v1"
kind: TransportServer
metadata:
  name: app-10
  labels:
    f5cr: "true"
spec:
  virtualServerAddress: "192.168.8.10"
  virtualServerPort: 80
  host: "host10.f5demo.cloud"
  virtualServerName: "app-10"
  mode: standard
  snat: auto
  pool:
    service: app-svc-10
    servicePort: 8080
    monitor:
      type: tcp
      interval: 3
      timeout: 10
---

```

#### Expected Result 
It should work regardless the order.

**Comment by trinaths** [comment]
Created [CONTCNTR-3826] for internal tracking.

**Comment by skenderidis** [comment]
Hi,

It was 3.41

**Comment by trinaths** [comment]
Fixed in 2.12.1
